facets_overview/components/facets_overview_table/facets-overview-table.ts

```diff
@@ -190,5 +204,12 @@ Polymer({
   },
   _getTableRowClass(numeric: boolean) {
     return numeric ? 'numeric-row' : 'categorical-row';
-  }
+  },
+  _shouldShowDataset(datasetIndex: number, datasetCheckboxes: boolean[]) {
+    if (datasetCheckboxes == null || datasetCheckboxes.length <= datasetIndex) {
```

Comment:
If I understand this code, it's equivalent to this one-liner:

```js
return datasetCheckboxes != null &&
  datasetCheckboxes.length > datasetIndex &&
  datasetCheckboxes[datasetIndex];
```

But I don't understand, under what circumstances would the `datasetIndex` extend beyond the length of `datasetCheckboxes`?

Reply:
simplified. there's no reason to check index against length of list. i was being overly protective given the code that creates the array.